src/Stan/Analysis.hs

```diff
@@ -23,7 +25,7 @@ data Analysis = Analysis
     { analysisModuleCount    :: !Int
     , analysisLinesOfCode    :: !Int
     , analysisUsedExtensions :: !Int
-    , analysisObservations   :: !()  -- TODO: use Observation type later
+    , analysisObservations   :: ![Observation]
```

Comment:
I can imagine that in future we would like to either keep a sized list here instead or to keep a number of observations along the way. That would be handy for report + filtering/ searching. What do you think? Can keep this in mind :)

Reply:
@vrom911 Nice use case for the sized list! I imagine we will find more places in our codebase where we can use sized list. This data type probably will change in the future, but I agree, let's keep it in mind :slightly_smiling_face: